jdaviz/configs/default/plugins/subset_plugin/tests/test_subset_plugin.py

```diff
@@ -168,3 +186,82 @@ def test_circle_recenter_linking(roi_class, subset_info, imviz_helper, image_2d_
+       SpectralRegion(7.004748267441649 * u.um, 7.3404016303483965 * u.um)], NewMode, 3, 1),
+     ((SpectralRegion(5.772486091213352 * u.um, 6.052963676101135 * u.um) +
+       SpectralRegion(6.494371022809778 * u.um, 6.724270682553864 * u.um) +
+       SpectralRegion(7.004748267441649 * u.um, 7.3404016303483965 * u.um)), OrMode, 1, 3)]
+)
+def test_import_spectral_region(cubeviz_helper, spectrum1d_cube, spec_regions, mode, len_subsets,
+                                len_subregions):
+    cubeviz_helper.load_data(spectrum1d_cube)
+    cubeviz_helper.app.get_tray_item_from_name('g-subset-plugin')
+    plg = cubeviz_helper.plugins['Subset Tools']._obj
+    plg._import_spectral_regions(spec_region=spec_regions, mode=mode)
+    subsets = cubeviz_helper.app.get_subsets()
+    assert len(subsets) == len_subsets
+    assert len(subsets['Subset 1']) == len_subregions
+
+
+test_subset_dict = {'Subset 1': [
+    {'name': 'TrueCircularROI',
+     'glue_state': 'AndState',
+     'region': CirclePixelRegion(center=PixCoord(x=20.6072998046875, y=23.749065399169922), radius=4.079286151278358),
+     'sky_region': None,
+     'subset_state': None},
+     {'name': 'RectangularROI',
+      'glue_state': 'OrState',
+      'region': RectanglePixelRegion(center=PixCoord(x=20.68303102318066, y=30.615620480255785), width=6.815691577924664, height=4.272523078699049, angle=0.0 * u.rad),
+      'sky_region': None,
+      'subset_state': None},
+    {'name': 'EllipticalROI',
+     'glue_state': 'AndNotState',
+     'region': EllipsePixelRegion(center=PixCoord(x=20.30438232421875, y=27.716407775878906), width=3.427145004272461, height=8.200210571289062, angle=0.0 * u.rad),
+     'sky_region': None,
+     'subset_state': None}],
+                       'Subset 2': [
+                           {'name': 'CircularAnnulusROI',
+                            'glue_state': 'RoiSubsetState',
+                            'region': CircleAnnulusPixelRegion(center=PixCoord(x=30.452190399169922, y=22.070573806762695), inner_radius=2.4145185947418213, outer_radius=4.829037189483643),
+                            'sky_region': None,
+                            'subset_state': None}],
+                       'Subset 3': SpectralRegion(6.939254409861322 * u.um, 7.224590119773996 * u.um)
+}
+
+test_subset_string = str(test_subset_dict)
+
+region = [{'name': 'TrueCircularROI',
+           'glue_state': 'RoiSubsetState',
+           'region': CirclePixelRegion(center=PixCoord(x=25.888399124145508, y=22.078184127807617),
+                                       radius=3.7648651881914112),
+           'sky_region': None,
+           'subset_state': ''},
+          {'name': 'TrueCircularROI',
+           'glue_state': 'OrState',
+           'region': CirclePixelRegion(center=PixCoord(x=19.990379333496094, y=30.782743453979492),
+                                       radius=2.756227940196304),
+           'sky_region': None,
+           'subset_state': ''}]
```

Comment:
aren't these internal formats?  Are we expecting users to write and pass regions in this format or do we need support for this internally?

Reply:
This is the format that `app.get_subsets` takes and it provides all relevant information for reconstructing subsets, so I thought it made sense to use as the format for importing as well.

Reply:
I thought we had planned to just support file or regions objects import, and leave more complex logic to importing successively and using the plugins API to change the combination mode?

Reply:
I can add support for that as well. My plan is to use the `import_region` method to support any kind of import, SpectralRegion, Region object, dictionary, list, and string. I can work on the plugin api next.